### ISSUE TITLE
Get gym solution for parser

### ISSUE THREAD

**Issue by unaimillan** [VadVergasov/CodeforcesApiPy#363]
Would be cool to add a parser function similar to `get_solution`, but for mashup contests created in gym. 
Implementation should be similar, except for `/gym/` in the url path

```python
def get_solution(self, contest_id, submit_id):
    solutionPage = self.session.get(
        "https://codeforces.com/gym/"
        + str(contest_id)
        + "/submission/"
        + str(submit_id)
    )
    if int(solutionPage.status_code) != 200:
        raise Exception("Returned not OK code " + str(solutionPage))
    tree2 = html.fromstring(solutionPage.text)
    code = tree2.xpath("//pre[@id='program-source-text']/text()")
    if len(code) == 0:
        raise ValueError("Incorrect contest_id or submit_id " + str(code))
    return code[0].replace("\r", "")
```
or just add one more parameter to `get_solution` with default value `gym=False`

**Comment by VadVergasov** [comment]
Ok, I'll look through it.

**Comment by VadVergasov** [comment]
I can see gym submissions aren't public, so you need to authenticate to grab the solution. So it's nearly impossible to create such a feature. Correct me if I'm mistaken.

**Comment by unaimillan** [comment]
Yes, this feature requires authorized access to api, so should be available only when api key and secret specified. 
Nevertheless, I am as a tutor could have plenty of different private contests for training and anyway will need to download student's solutions for plagiarism and code style checks.

**Comment by unaimillan** [comment]
Ahh, understood the problem, as parser requests are made directly bypassing API, this feature requires access to contest manager account(
